extension_service/datastore/providers/cloudsql.py

```diff
@@ -0,0 +1,464 @@
+                    user=f"{config.user}",
+                    password=f"{config.password}",
+                    db=f"{config.database}",
+                )
+            return conn
+
+        pool = create_async_engine(
+            "postgresql+asyncpg://",
+            async_creator=getconn,
+        )
+        if pool is None:
+            raise TypeError("pool not instantiated")
+        return cls(pool)
+
+    async def initialize_data(
+        self,
+        airports: list[models.Airport],
+        amenities: list[models.Amenity],
+        flights: list[models.Flight],
+    ) -> None:
+        async with self.__pool.connect() as conn:
+            # If the table already exists, drop it to avoid conflicts
+            await conn.execute(sqlalchemy.text("DROP TABLE IF EXISTS airports CASCADE"))
+            # Create a new table
+            await conn.execute(
+                sqlalchemy.text(
+                    """
+                    CREATE TABLE airports(
+                      id INT PRIMARY KEY,
+                      iata TEXT,
+                      name TEXT,
+                      city TEXT,
+                      country TEXT
+                    )
+                    """
+                )
+            )
+            # Insert all the data
+            await conn.execute(
+                sqlalchemy.text(
+                    """INSERT INTO airports VALUES (:id, :iata, :name, :city, :country)"""
+                ),
+                [
+                    {
+                        "id": a.id,
+                        "iata": a.iata,
+                        "name": a.name,
+                        "city": a.city,
+                        "country": a.country,
+                    }
+                    for a in airports
+                ],
+            )
+
+            await conn.execute(sqlalchemy.text("CREATE EXTENSION IF NOT EXISTS vector"))
+            # If the table already exists, drop it to avoid conflicts
+            await conn.execute(
+                sqlalchemy.text("DROP TABLE IF EXISTS amenities CASCADE")
+            )
+            # Create a new table
+            await conn.execute(
+                sqlalchemy.text(
+                    """
+                    CREATE TABLE amenities(
+                      id INT PRIMARY KEY,
+                      name TEXT,
+                      description TEXT,
+                      location TEXT,
+                      terminal TEXT,
+                      category TEXT,
+                      hour TEXT,
+                      content TEXT NOT NULL,
+                      embedding vector(768) NOT NULL
+                    )
+                    """
+                )
+            )
+            # Insert all the data
+            await conn.execute(
+                sqlalchemy.text(
+                    """INSERT INTO amenities VALUES (:id, :name, :description, :location, :terminal, :category, :hour, :content, :embedding)"""
+                ),
```

Comment:
super-nit: we're using `sqlalchemy.text` enough it's probably cleaner to `from sqlalchemy import text` instead